.github/workflows/framework-e2e.yml

```diff
@@ -190,54 +200,51 @@ jobs:
               from sklearn.datasets import load_iris
               Path('data').mkdir(exist_ok=True)
               load_iris(as_frame=True)['data'].to_csv('./data/client.csv')
+            pytorch_cpu: false
 
     name: Framework / ${{ matrix.directory }}
 
     defaults:
       run:
         working-directory: framework/e2e/${{ matrix.directory }}
-
     steps:
       - uses: actions/checkout@v4
-      - name: Set up Python
-        if: ${{ needs.changes.outputs.framework == 'true' }}
-        uses: actions/setup-python@v5
+      - name: Install uv and set the Python version
+        uses: astral-sh/setup-uv@v6
         with:
-          python-version: ${{ (matrix.directory == 'e2e-fastai') && '3.10' || '3.9' }}
-      - name: Install build tools
-        if: ${{ needs.changes.outputs.framework == 'true' }}
-        run: |
-            python -m pip install -U pip==23.3.1
-        shell: bash
+          python-version: "3.10"
       # Using approach described here for Python location caching:
       # https://blog.allenai.org/python-caching-in-github-actions-e9452698e98d
-      - name: Cache Python location
-        if: ${{ needs.changes.outputs.framework == 'true' }}
-        id: cache-restore-python
-        uses: actions/cache/restore@v4
-        with:
-          path: ${{ env.pythonLocation }}
-          key: pythonloc-${{ runner.os }}-${{ matrix.directory }}-${{ env.pythonLocation }}-${{ hashFiles(format('./framework/e2e/{0}/pyproject.toml', matrix.directory)) }}
-      - name: Install dependencies
+      # - name: Cache Python location
+      #   if: ${{ needs.changes.outputs.framework == 'true' }}
+      #   id: cache-restore-python
+      #   uses: actions/cache/restore@v4
+      #   with:
+      #     path: ${{ env.pythonLocation }}
+      #     key: pythonloc-${{ runner.os }}-${{ matrix.directory }}-${{ env.pythonLocation }}-${{ hashFiles(format('./framework/e2e/{0}/pyproject.toml', matrix.directory)) }}
```

Comment:
Why is this commented out? We tend to delete thing instead of commenting them out